### PR TITLE
raise LoadError when a non-existent file or directory is specified to the test runner

### PR DESCRIPTION
**Pull request by y-yagi**
Currently, if a file or directory that does not exist was specified in the test runner, that argument is ignored.
This commit has been modified to cause an error if there is no file or directory.

In addition, there is a problem that does not pass correctly the argument to test runner from the rake,  thereby for now test will fail,  was modified together.

### CODE REVIEW

**Comment by senny** [comment]
I remember wanting to change the same a couple weeks ago. I think after some back and forth with @kaspth we let it be for the moment. Maybe he remembers specifics :grin:

**Comment by kaspth** [comment]
I think it was running several rake tasks in one command: `bin/rake db:migrate test`. Because we can’t tell if the thing missing is a rake task, why can’t really fix this.

Perhaps checking if it’s a rake task before raising could do?

> Den 07/09/2015 kl. 09.07 skrev Yves Senn <notifications@github.com>:
> 
> I remember wanting to change the same a couple weeks ago. I think after some back and forth with @kaspth <https://github.com/kaspth> we let it be for the moment. Maybe he remembers specifics 
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/rails/pull/21519#issuecomment-138211315>.
> 

--
Kasper

File: railties/lib/rails/test_unit/minitest_plugin.rb [hunk 1]
```diff
@@ -26,7 +26,11 @@ def self.plugin_rails_options(opts, options)
       options[:full_backtrace] = true
     end
 
-    options[:patterns] = opts.order!
+    if self.test_patterns
+      options[:patterns] = self.test_patterns
+    else
+      options[:patterns] = opts.order!
```
Comment (kaspth) [inline railties/lib/rails/test_unit/minitest_plugin.rb]: Should just be: `options[:patterns] = test_patterns || opts.order!`

File: railties/lib/rails/test_unit/minitest_plugin.rb [hunk 2]
```diff
@@ -46,6 +50,7 @@ def self.plugin_rails_init(options)
 
   mattr_accessor(:run_with_autorun)         { false }
   mattr_accessor(:run_with_rails_extension) { false }
+  mattr_accessor(:test_patterns) { nil }
```
Comment (kaspth) [inline railties/lib/rails/test_unit/minitest_plugin.rb]: Let's get rid of the test patterns and add a `rake_run` method, that looks like:

```
# Running several Rake tasks in a single command would trip up the runner,
# as the patterns would also contain the other Rake tasks.
def self.rake_run(patterns) # :nodoc:
  @rake_patterns = patterns
  run
end

def self.plugin_rails_init(options)
  # Other code

  unless run_with_autorun
    ::Rails::TestRequirer.require_files @rake_patterns || options[:patterns]
  end

  # More initialization...
end
```

File: railties/lib/rails/test_unit/minitest_plugin.rb (hunk 1 above)
Comment (kaspth) [inline railties/lib/rails/test_unit/minitest_plugin.rb]: Disregard this and go with my other comment :grin:

**Comment by kaspth** [comment]
Awesome, thanks for working on this :+1: There's a few things I'd like you to address, but otherwise it should be smooth sailing :grin:

**Comment by y-yagi** [comment]
@kaspth Thanks for reviewing!  I updated commit.

**Comment by kaspth** [comment]
Thanks!

**Comment by y-yagi** [comment]
Thanks!

**Pull request merged**